### ISSUE TITLE
Rotation

### ISSUE THREAD

**Issue by xogeny** [x3dom/x3dom#565]
I notice in the documentation, the types of rotation are not consistent or even clearly specified.  For example, the `rotation` attribute on a `Transform` element is (presumably) axis-angle (sure looks that way).  [The documentation](http://doc.x3dom.org/developer/x3dom/nodeTypes/Transform.html) only says "The rotation field specifies a rotation of the coordinate system."  The type is just listed as `SFRotation` with no link to any further explanation.

This is very confusing because this rotation could be Euler angles, axis-angle, quaternions and who knows what else (I'm not an expert in 3D stuff).

In addition, the [Quaternion documentation](http://doc.x3dom.org/developer/x3dom/fields/Quaternion.html) doesn't include any methods to convert between representations.  This seems odd.  Not even a pointer to documentation about how to do it.  The fact that most of the uses with respect to `SFMatrix4f` seem to be quaternions seems odd given that `rotation` is axis-angle.

Bottom line is I think more specificity would help here.

**Comment by mlimper** [comment]
> The documentation only says "The rotation field specifies a rotation of the coordinate system." The type is just listed as SFRotation with no link to any further explanation.

That's true, SFRotation should be specified somewhere. However, for what it's worth, let me just add that X3DOM tries to implement the X3D standard as close as possible, at least for such standard nodes like Transform. Therfore, you might look things like this up in the X3D specification, where it is really described pretty well in all details:

http://www.web3d.org/documents/specifications/19775-1/V3.2/Part01/components/group.html#Transform


> In addition, the Quaternion documentation doesn't include any methods to convert between representations. This seems odd. Not even a pointer to documentation about how to do it. The fact that most of the uses with respect to SFMatrix4f seem to be quaternions seems odd given that rotation is axis-angle.

Good point, I believe this is even more serious, as it seems there is no documentation at all right now for Quaternion. Although these methods exist:
https://github.com/x3dom/x3dom/blob/master/src/fields.js#L1742

I think we just need to fill in jsdoc comments, like it has already been done for SFMatrix, for example:
Code with comments: https://github.com/x3dom/x3dom/blob/master/src/fields.js#L69
Resulting documentation: http://doc.x3dom.org/developer/x3dom/fields/SFMatrix4f.html

volunteers are very welcome :-)

**Comment by Sheeban-Wasi** [comment]
Hi @mlimper could you please suggest how can I access the setRotate method in sfmatrix4f() ? as what I am trying to do is:

we are rotating the hit point around the origin 
positive  90 deg around the positive x axis

back to 'vertical' 
then translating the offset back 

so out hitpoint SFVEC
has to go through at least 1 if not two 4x4 matrix multiplies - likely right multiplies

is first rotated by '1 0 0 1.57079632679' 

then translated by '440000, 3670000, 0'

**Pseudocode example**

(still pseudocode, but the operations for example)

```
hitp = new SFVec3f (x, y, z); // our hit point from x3d 

myRot = new SFRotation(1 0 0 1.57079632679);
myTrans =   new SFVec3f ( 440000, 3670000, 0  );

rotMat = new SFMatrix4f ();
rotMat.setRotation (myRot);

transMat = new SFMatrix4f ();
transMat.setTranslation(myTrans);

locR = sfmatrix4f.multMatrixVec(sfvec3f);
locT = transMat.multVecMatrix(locR); 

```
Thanks in advance!
